### PR TITLE
chore(ui): handle trace not being available without exceptions

### LINKED ISSUE #33339: [internal]: hide trace viewer error when fetching trace JSON
> Recently i came across an error (Uncaught SyntaxError: Unexpected end of JSON input at JSON.parse ) that's triggered from
> below code inside the tracer console.
> 
> 
> `    constructor(t) {
>         M(this, "_entriesPromise");
>         M(this, "_traceURL");
>         this._traceURL = t,
>         this._entriesPromise = fetch("/trace/file?path=" + encodeURIComponent(t)).then(async e=>{
>             const r = JSON.parse(await e.text())
>               , s = new Map;
>             for (const i of r.entries)
>                 s.set(i.name, i.path);
>             return s
>         }
>         `

 _Originally posted by @miordache93 in [#33328](https://github.com/microsoft/playwright/issues/33328#issuecomment-2442337912)_

This error is expected. Let's catch it and not surface it in the console.

### PR DESCRIPTION
**Pull request by Skn0tt**
The control flow for loading live traces is currently based on exceptions, and it's hard to follow. It's expected that the live trace only becomes available after a couple retries, but the code doesn't reflect that.

This PR adds some handling for 404s so we represent the "trace not yet available" state with 404 responses instead of 
errors. It also renames the `counter` state to `rerender` to make the intent clearer.

Console before this change:

Screenshot 2024-11-04 at 16 01 58

Console after this change:

Screenshot 2024-11-04 at 16 02 46

Closes https://github.com/microsoft/playwright/issues/33339.

### CODE REVIEW

File: packages/trace-viewer/src/sw/main.ts [hunk 1]
```diff
@@ -54,6 +54,8 @@ async function loadTrace(traceUrl: string, traceFileName: string | null, clientI
     const backend = traceUrl.endsWith('json') ? new FetchTraceModelBackend(traceUrl) : new ZipTraceModelBackend(traceUrl, fetchProgress);
     await traceModel.load(backend, unzipProgress);
   } catch (error: any) {
+    if (error?.message === 'trace not found')
```
Comment (pavelfeldman) [inline packages/trace-viewer/src/sw/main.ts:57]: Looks like you are mostly addressing the malformed (empty) JSON in live mode. Should we check that at parse time and return empty trac instead?

Comment (Skn0tt) [reply packages/trace-viewer/src/sw/main.ts:57]: That's a nice idea as well, but I think it'd be more complex. Just one example, we're treating empty traces as an error while parsing:

https://github.com/microsoft/playwright/blob/26a7dd4dd468bd15e0aa51dd5f190b7b72d8df4c/packages/trace-viewer/src/sw/traceModel.ts#L52-L53

**Comment by Skn0tt** [comment]
closing this as per discussion yesterday. we have lots of errors in the console already, some non removable anyways. no point in battling one of them if we can't battle them all.
